### PR TITLE
Chapter1/exterminator

### PR DESCRIPTION
**Pull request by ghost**
I hope I found all the errors and make comments where I think they are and why

### CODE REVIEW

**Comment by RoelofWobben** [comment]
I close this one because its against the wrong branch

File: Src/BootCamp.Chapter/Check.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Completing homework 3 before looking at this is HIGHLY recommended.
+// Try to look at the code in GitHub first. Try to find the mistakes that
+// were made without help or tools first.
+// After that try to find every single thing that seems off.
+// Have fun! :)
+namespace BootCamp.Chap
+{
+    class Checks
+    {
+        public string PromptString(String Message) // this must be string because you want the type string not the class string.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Check.cs]: Codacy Issue found: [Make 'PromptString' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Check.cs  1
- Src/BootCamp.Chapter/lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581&bid=16184928)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿// This file is for hunting bugs only.
+// Completing homework 3 before looking at this is HIGHLY recommended.
+// Try to look at the code in GitHub first. Try to find the mistakes that
+// were made without help or tools first.
+// After that try to find every single thing that seems off.
+// Have fun! :)
+using System;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+using BootCamp.Chap;
+
+namespace BootCamp.Chapter
+{
+    internal class lesson3   // class name must begin with a capital character. 
+    {
+        public static int demo(string count)  // name must begin with a capital character.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename method 'demo' to match pascal case naming rules, consider using 'Demo'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            int count = count;  // on initialisation a variable can never point to itself. because at that point it null and it makes no sense 
+            do
+            {
+                ProcPrsn(); // Name is wrong. so the compiler cannot find it. 
+                // count is never updated so this will be a endless loop. 
+            } while (iteration < count);   // Iteration is never initialized so it does not exist. 
+        }
+
+        private static void PrcPrsn()
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused private method 'PrcPrsn'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/Check.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            return Console.ReadLine(); 
+        }
+
+        public int PromptInt(String message) // this must be string because you want the type string not the class string.
+        {
+            // Lesson3 must be written with a capital first character because you want to use the class Lesson3. 
+            lesson3.ReadMyInput();
+        }
+
+        public float PromptFloat(String message) // this must be string because you want the type string not the class string.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Check.cs]: Codacy Issue found: [Make 'PromptFloat' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+// Try to look at the code in GitHub first. Try to find the mistakes that
+// were made without help or tools first.
+// After that try to find every single thing that seems off.
+// Have fun! :)
+using System;
+using BootCamp.Chap;
+
+namespace BootCamp.Chapter
+{
+    internal class lesson3   // class name must begin with a capital character.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename class 'lesson3' to match pascal case naming rules, consider using 'Lesson3'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/Check.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    class Checks
+    {
+        public string PromptString(String Message) // this must be string because you want the type string not the class string.
+        {
+           // No using statement for System so Console is unknown.
+           // No call to the own function so it is not testable. 
+            return Console.ReadLine(); 
+        }
+
+        public int PromptInt(String message) // this must be string because you want the type string not the class string.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Check.cs]: Codacy Issue found: [Make 'PromptInt' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            return int.ReadFromConsole();  // Int does not have a possibility to read things from console. So I think it should be Console. or Clas3 but then the function need to be written.
+        }
+
+        private static float AsFloat()
+        {
+            return (float)ReadMyInput();
+        }
+
+        private void calcBMI()
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename method 'calcBMI' to match pascal case naming rules, consider using 'CalcBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        private static void PrcPrsn()
+        {
+            WriteLine("What's ya name, mate?");  // A Console is mising so the compiler thinks its a function in Class3 
+            Checks.PromptString(name);   // name is is never initialized so it does not exist
+            WriteLine("What is your weight and height?"); // A Console. is mising so the compiler thinks its a function in Class3
+            float weightAndHeight = AsFloat();
+            WriteLine("And your age?");  // A Console. is mising so the compiler thinks its a function in Class3
+            int age = ReadMyInput();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused local variable 'age'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+                // count is never updated so this will be a endless loop. 
+            } while (iteration < count);   // Iteration is never initialized so it does not exist. 
+        }
+
+        private static void PrcPrsn()
+        {
+            WriteLine("What's ya name, mate?");  // A Console is mising so the compiler thinks its a function in Class3 
+            Checks.PromptString(name);   // name is is never initialized so it does not exist
+            WriteLine("What is your weight and height?"); // A Console. is mising so the compiler thinks its a function in Class3
+            float weightAndHeight = AsFloat();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused local variable 'weightAndHeight'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 8 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused private method 'calcBMI'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Make 'calcBMI' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

**Comment by ellmaster** [comment]
Good catches! :) I think there is more to detect.

**Comment by RoelofWobben** [comment]
oke,  I  think I know some more. Is there a way I can change the branch problem ?

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Check.cs  1
- Src/BootCamp.Chapter/lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581&bid=16184928)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 6 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename class 'lesson3' to match pascal case naming rules, consider using 'Lesson3'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            return int.ReadFromConsole();  // Int does not have a possibility to read things from console. So I think it should be Console. or Clas3 but then the function need to be written.
+        }
+
+        private static float AsFloat()
+        {
+            return (float)ReadMyInput();
+        }
+
+        private void calcBMI()   // class is missing a static so a class initialisation is needed to use this function.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Make 'calcBMI' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 4 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused private method 'PrcPrsn'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 10 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused local variable 'weightAndHeight'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/Check.cs [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    class Checks // class is now private so the functions could not be used outside the class and all functions must be static so you do not have to initialize a class to use them.
+    {
+        public string PromptString(String Message) // this must be string because you want the type string not the class string.
+        {
+           // No using statement for System so Console is unknown.
+           // No call to the own function so it is not testable. 
+            return Console.ReadLine(); 
+        }
+
+        public int PromptInt(String message) // this must be string because you want the type string not the class string.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Check.cs]: Codacy Issue found: [Make 'PromptInt' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 9 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused local variable 'age'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 2 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/Check.cs [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+// Completing homework 3 before looking at this is HIGHLY recommended.
+// Try to look at the code in GitHub first. Try to find the mistakes that
+// were made without help or tools first.
+// After that try to find every single thing that seems off.
+// Have fun! :)
+namespace BootCamp.Chap
+{
+    class Checks // class is now private so the functions could not be used outside the class and all functions must be static so you do not have to initialize a class to use them.
+    {
+        public string PromptString(String Message) // this must be string because you want the type string not the class string.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Check.cs]: Codacy Issue found: [Make 'PromptString' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 11 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove the unused private method 'calcBMI'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename method 'calcBMI' to match pascal case naming rules, consider using 'CalcBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 3 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename method 'demo' to match pascal case naming rules, consider using 'Demo'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

File: Src/BootCamp.Chapter/Check.cs [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+            return Console.ReadLine(); 
+        }
+
+        public int PromptInt(String message) // this must be string because you want the type string not the class string.
+        {
+            // Lesson3 must be written with a capital first character because you want to use the class Lesson3. 
+            lesson3.ReadMyInput();
+        }
+
+        public float PromptFloat(String message) // this must be string because you want the type string not the class string.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Check.cs]: Codacy Issue found: [Make 'PromptFloat' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885581)

**Comment by RoelofWobben** [comment]
Closed this one . There is a new PR with the right branch

**Pull request closed without merging**